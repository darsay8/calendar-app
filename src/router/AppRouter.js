import { BrowserRouter as Router, Routes, Route, Navigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import LoginScreen from '../components/auth/LoginScreen';
import CalendarScreen from '../components/calendar/CalendarScreen';
import { startChecking } from '../redux/actions/auth';
import { useEffect } from 'react';

export const AppRouter = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);

  return (
    <Router>
      <div>
        <Routes>
          <Route path="/login" element={<LoginScreen />} />

          <Route path="/" element={<CalendarScreen />} />

          <Route path="*" element={<Navigate replace to="/" />} />
        </Routes>
      </div>
    </Router>
  );
};
