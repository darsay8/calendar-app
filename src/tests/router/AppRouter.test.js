import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { AppRouter } from '../../router/AppRouter';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe(' <AppRouter/> Test', () => {
  test('should display loading...', () => {
    const initState = {
      auth: {
        checking: true,
      },
    };
    const store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>,
    );

    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h5').exists()).toBe(true);
  });

  test('should display <PublicRoute/>', () => {
    const initState = {
      auth: {
        checking: false,
        uid: null,
      },
    };
    const store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.login-screen').exists()).toBe(true);
  });

  test('should display <PrivateRoute/>', () => {
    const initState = {
      ui: {
        modalOpen: false,
      },
      calendar: {
        events: [],
        activeEvent: null,
      },
      auth: {
        checking: false,
        uid: '123',
        name: 'someone',
      },
    };
    const store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.calendar-screen').exists()).toBe(true);
  });
});
