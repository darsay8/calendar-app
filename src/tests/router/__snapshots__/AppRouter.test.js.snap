// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <AppRouter/> Test should display <PrivateRoute/> 1`] = `
<div>
  <div
    className="calendar-screen"
  >
    <nav
      className="navbar navbar-dark bg-dark justify-content-between mb-4"
    >
      <div
        className="container"
      >
        <span
          className="navbar-brand"
        >
          someone
        </span>
        <button
          className="btn btn-outline-danger"
          onClick={[Function]}
        >
          <i
            className="fas fa-sign-out-alt"
          />
          <span>
             Logout
          </span>
        </button>
      </div>
    </nav>
    <div
      className="container calendar"
    >
      <div
        className="rbc-calendar"
      >
        <div
          className="rbc-toolbar"
        >
          <span
            className="rbc-btn-group"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              Today
            </button>
            <button
              onClick={[Function]}
              type="button"
            >
              Back
            </button>
            <button
              onClick={[Function]}
              type="button"
            >
              Next
            </button>
          </span>
          <span
            className="rbc-toolbar-label"
          >
            April 2022
          </span>
          <span
            className="rbc-btn-group"
          >
            <button
              className="rbc-active"
              key="month"
              onClick={[Function]}
              type="button"
            >
              Month
            </button>
            <button
              className=""
              key="week"
              onClick={[Function]}
              type="button"
            >
              Week
            </button>
            <button
              className=""
              key="day"
              onClick={[Function]}
              type="button"
            >
              Day
            </button>
            <button
              className=""
              key="agenda"
              onClick={[Function]}
              type="button"
            >
              Agenda
            </button>
          </span>
        </div>
        <div
          aria-label="Month View"
          className="rbc-month-view"
          role="table"
        >
          <div
            className="rbc-row rbc-month-header"
            role="row"
          >
            <div
              className="rbc-header"
              key="header_0"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Sun
              </span>
            </div>
            <div
              className="rbc-header"
              key="header_1"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Mon
              </span>
            </div>
            <div
              className="rbc-header"
              key="header_2"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Tue
              </span>
            </div>
            <div
              className="rbc-header"
              key="header_3"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Wed
              </span>
            </div>
            <div
              className="rbc-header"
              key="header_4"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Thu
              </span>
            </div>
            <div
              className="rbc-header"
              key="header_5"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Fri
              </span>
            </div>
            <div
              className="rbc-header"
              key="header_6"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Sat
              </span>
            </div>
          </div>
          <div
            className="rbc-month-row"
            role="rowgroup"
          >
            <div
              className="rbc-row-bg"
            >
              <div
                className="rbc-day-bg rbc-off-range-bg"
              />
              <div
                className="rbc-day-bg rbc-off-range-bg"
              />
              <div
                className="rbc-day-bg rbc-off-range-bg"
              />
              <div
                className="rbc-day-bg rbc-off-range-bg"
              />
              <div
                className="rbc-day-bg rbc-off-range-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
            </div>
            <div
              className="rbc-row-content"
              role="row"
            >
              <div
                className="rbc-row "
              >
                <div
                  className="rbc-date-cell rbc-off-range"
                  key="header_0"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    27
                  </button>
                </div>
                <div
                  className="rbc-date-cell rbc-off-range"
                  key="header_1"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    28
                  </button>
                </div>
                <div
                  className="rbc-date-cell rbc-off-range"
                  key="header_2"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    29
                  </button>
                </div>
                <div
                  className="rbc-date-cell rbc-off-range"
                  key="header_3"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    30
                  </button>
                </div>
                <div
                  className="rbc-date-cell rbc-off-range"
                  key="header_4"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    31
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_5"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    01
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_6"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    02
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="rbc-month-row"
            role="rowgroup"
          >
            <div
              className="rbc-row-bg"
            >
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
            </div>
            <div
              className="rbc-row-content"
              role="row"
            >
              <div
                className="rbc-row "
              >
                <div
                  className="rbc-date-cell"
                  key="header_0"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    03
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_1"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    04
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_2"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    05
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_3"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    06
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_4"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    07
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_5"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    08
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_6"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    09
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="rbc-month-row"
            role="rowgroup"
          >
            <div
              className="rbc-row-bg"
            >
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
            </div>
            <div
              className="rbc-row-content"
              role="row"
            >
              <div
                className="rbc-row "
              >
                <div
                  className="rbc-date-cell"
                  key="header_0"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    10
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_1"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    11
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_2"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    12
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_3"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    13
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_4"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    14
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_5"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    15
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_6"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    16
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="rbc-month-row"
            role="rowgroup"
          >
            <div
              className="rbc-row-bg"
            >
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg rbc-today"
              />
              <div
                className="rbc-day-bg"
              />
            </div>
            <div
              className="rbc-row-content"
              role="row"
            >
              <div
                className="rbc-row "
              >
                <div
                  className="rbc-date-cell"
                  key="header_0"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    17
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_1"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    18
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_2"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    19
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_3"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    20
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_4"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    21
                  </button>
                </div>
                <div
                  className="rbc-date-cell rbc-now rbc-current"
                  key="header_5"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    22
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_6"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    23
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="rbc-month-row"
            role="rowgroup"
          >
            <div
              className="rbc-row-bg"
            >
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
              <div
                className="rbc-day-bg"
              />
            </div>
            <div
              className="rbc-row-content"
              role="row"
            >
              <div
                className="rbc-row "
              >
                <div
                  className="rbc-date-cell"
                  key="header_0"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    24
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_1"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    25
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_2"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    26
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_3"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    27
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_4"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    28
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_5"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    29
                  </button>
                </div>
                <div
                  className="rbc-date-cell"
                  key="header_6"
                  role="cell"
                >
                  <button
                    className="rbc-button-link"
                    onClick={[Function]}
                    role="cell"
                    type="button"
                  >
                    30
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="btn btn-primary fab"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </button>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
  </div>
</div>
`;

exports[` <AppRouter/> Test should display <PublicRoute/> 1`] = `
<div>
  <div
    className="login-screen"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="forms-group"
        >
          <div
            className="col-md-6 login-form-1"
          >
            <h3>
              Login
            </h3>
            <form
              className="form-horizontal"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="loginEmail"
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="loginPassword"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="btnSubmit"
                  type="submit"
                  value="Login"
                />
              </div>
            </form>
          </div>
          <div
            className="col-md-6 login-form-2"
          >
            <h3>
              Register
            </h3>
            <form
              className="form-horizontal"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control"
                  name="registerName"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="registerEmail"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="registerPassword"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  name="registerConfirmPassword"
                  onChange={[Function]}
                  placeholder="Confirm password"
                  type="password"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="btnSubmit"
                  type="submit"
                  value="Register"
                />
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
